File: Modelo/OrdenCompra.py
from enum import Enum

class Estado(Enum):
    PENDIENTE = 'PENDIENTE'
    RECIBIDO = 'RECIBIDO'

class TipoPago(Enum):
    CONTADO = 'CONTADO'
    CREDITO = 'CREDITO'

class ItemOrden:
    def __init__(self, producto=None, cantidadSolicitada=0.0, precioCosto=0.0):
        self._producto = producto
        self._cantidadSolicitada = cantidadSolicitada
        self._precioCosto = precioCosto

    def getProducto(self):
        return self._producto

    def setProducto(self, producto):
        self._producto = producto

    def getCantidadSolicitada(self):
        return self._cantidadSolicitada

    def setCantidadSolicitada(self, cantidadSolicitada):
        self._cantidadSolicitada = cantidadSolicitada

    def getPrecioCosto(self):
        return self._precioCosto

    def setPrecioCosto(self, precioCosto):
        self._precioCosto = precioCosto

class OrdenCompra:
    ItemOrden = ItemOrden

    def __init__(self, id=0, proveedor=None, items=None, total=0.0, tipoPago=None, estado=None, fecha=None):
        self._id = id
        self._proveedor = proveedor
        self._items = items if items is not None else []
        self._total = total
        self._tipoPago = tipoPago
        self._estado = estado
        self._fecha = fecha

    def getId(self):
        return self._id

    def setId(self, id):
        self._id = id

    def getProveedor(self):
        return self._proveedor

    def setProveedor(self, proveedor):
        self._proveedor = proveedor

    def getItems(self):
        return self._items

    def setItems(self, items):
        self._items = items

    def getTotal(self):
        return self._total

    def setTotal(self, total):
        self._total = total

    def getTipoPago(self):
        return self._tipoPago

    def setTipoPago(self, tipoPago):
        self._tipoPago = tipoPago

    def getEstado(self):
        return self._estado

    def setEstado(self, estado):
        self._estado = estado

    def getFecha(self):
        return self._fecha

    def setFecha(self, fecha):
        self._fecha = fecha

File: Modelo/OrdenCompra.cs
using System;
using System.Collections.Generic;

namespace punto_de_venta_C_.Modelo
{
    public class OrdenCompra
    {
        public enum Estado
        {
            PENDIENTE, RECIBIDO
        }

        public enum TipoPago
        {
            CONTADO, CREDITO
        }

        private int id;
        private Proveedor proveedor;
        private List<ItemOrden> items;
        private double total;
        private TipoPago tipoPago;
        private Estado estado;
        private DateTime fecha;

        public OrdenCompra(int id, Proveedor proveedor, List<ItemOrden> items, double total, TipoPago tipoPago, Estado estado, DateTime fecha)
        {
            this.id = id;
            this.proveedor = proveedor;
            this.items = items;
            this.total = total;
            this.tipoPago = tipoPago;
            this.estado = estado;
            this.fecha = fecha;
        }

        public int getId() { return id; }
        public void setId(int id) { this.id = id; }
        public Proveedor getProveedor() { return proveedor; }
        public void setProveedor(Proveedor proveedor) { this.proveedor = proveedor; }
        public List<ItemOrden> getItems() { return items; }
        public void setItems(List<ItemOrden> items) { this.items = items; }
        public double getTotal() { return total; }
        public void setTotal(double total) { this.total = total; }
        public TipoPago getTipoPago() { return tipoPago; }
        public void setTipoPago(TipoPago tipoPago) { this.tipoPago = tipoPago; }
        public Estado getEstado() { return estado; }
        public void setEstado(Estado estado) { this.estado = estado; }
        public DateTime getFecha() { return fecha; }
        public void setFecha(DateTime fecha) { this.fecha = fecha; }

        public bool isPendiente()
        {
            return estado == Estado.PENDIENTE;
        }

        public string getFolioCorto()
        {
            return "OC-" + string.Format("{0:0000}", id % 10000);
        }

        public class ItemOrden
        {
            private Producto producto;
            private double cantidadSolicitada;
            private double precioCosto;

            public ItemOrden(Producto producto, double cantidadSolicitada, double precioCosto)
            {
                this.producto = producto;
                this.cantidadSolicitada = cantidadSolicitada;
                this.precioCosto = precioCosto;
            }

            public Producto getProducto() { return producto; }
            public void setProducto(Producto producto) { this.producto = producto; }
            public double getCantidadSolicitada() { return cantidadSolicitada; }
            public void setCantidadSolicitada(double cantidadSolicitada) { this.cantidadSolicitada = cantidadSolicitada; }
            public double getPrecioCosto() { return precioCosto; }
            public void setPrecioCosto(double precioCosto) { this.precioCosto = precioCosto; }

            public double getSubtotal()
            {
                return precioCosto * cantidadSolicitada;
            }
        }
    }
}
